src/hypothesis/internal/cathetus.py

```diff
@@ -0,0 +1,63 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2018 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+from sys import float_info
+from math import fabs, sqrt, isinf, isnan
+
+
+def cathetus(h, a):
+    """Given the lengths of the hypotenuse and a side of a right triangle,
+    return the length of the other side.
+
+    A companion to the C99 hypot() function.  Some care is needed to avoid
+    underflow in the case of small arguments, and overflow in the case of
+    large arguments as would occur for the naive implementation as
+    sqrt(h*h - a*a).  The behaviour with respect the non-finite arguments
+    (NaNs and infinities) is designed to be as consistent as possible with
+    the C99 hypot() specifications.
+
+    Based on the C99 implementation https://github.com/jjgreen/cathetus
+    """
+    if isnan(h):
+        return float(u'nan')
+
+    if isinf(h):
+        if isinf(a):
+            return float(u'nan')
+        else:
+            # Includes the case when a is NaN, deliberately, since the
```

Comment:
nit: this is why I dislike single-letter variable names, because this could be easily mistaken for a typo!

Either improve the names, or maybe wrap this in quotes to be clear 'a' is a variable? i.e.

```python
# Includes the case when `a` is NaN, deliberately, since the
```